### PR TITLE
[ABI-Break][SYCL] ABI-neutralize device:: and platform::has_extension

### PR DESCRIPTION
**Pull request by bso-intel**
By passing a `detail::string_view` instead of `std::string`, we can make some APIs ABI-neutral.

### CODE REVIEW

**Comment by bso-intel** [comment]
@sergey-semenov 
friendly ping

**Comment by bso-intel** [comment]
@sergey-semenov 
ping

**Review by sergey-semenov — APPROVED** [review]

**Comment by bso-intel** [comment]
@intel/llvm-gatekeepers 
Please merge.

**Review by steffenlarsen — CHANGES_REQUESTED** [review]

File: sycl/include/sycl/device.hpp [hunk 1]
```diff
@@ -236,7 +236,10 @@ class __SYCL_EXPORT device : public detail::OwnerLessBase<device> {
   /// \param extension_name is a name of queried extension.
   /// \return true if SYCL device supports the extension.
   __SYCL2020_DEPRECATED("use device::has() function with aspects APIs instead")
-  bool has_extension(const std::string &extension_name) const;
+  bool has_extension(const std::string &extension_name) const {
+    return has_extension(detail::string_view{extension_name});
+  }
+  bool has_extension(detail::string_view extension_name) const;
```
[the remark this answers is not in the document]
Comment (bso-intel) [reply sycl/include/sycl/device.hpp]: Done. Thanks

File: sycl/include/sycl/platform.hpp [hunk 2]
```diff
@@ -157,7 +157,10 @@ class __SYCL_EXPORT platform : public detail::OwnerLessBase<platform> {
   /// \return true if specified extension is supported by this SYCL platform.
   __SYCL2020_DEPRECATED(
       "use platform::has() function with aspects APIs instead")
-  bool has_extension(const std::string &ExtensionName) const;
+  bool has_extension(const std::string &ExtensionName) const {
+    return has_extension(detail::string_view{ExtensionName});
+  }
+  bool has_extension(detail::string_view ExtensionName) const;
```
[the remark this answers is not in the document]
Comment (bso-intel) [reply sycl/include/sycl/platform.hpp]: Done. Thanks

**Review by steffenlarsen — APPROVED** [review]

**Comment by bso-intel** [comment]
@intel/llvm-gatekeepers 
Please merge.

**Pull request merged**